Cargo.toml

```diff
@@ -9,9 +9,10 @@ members = [
 resolver = "2"
 
 [profile.release]
-opt-level = 3
+opt-level = "s"
```

Comment:
Would be great to have a comment about what this is or where you can learn more? (or this is just standard rust and you should disregard)

Reply:
good callout and added the link but adding more details here.

So what I found with default vs "s" is a 13% reduction in wasm size and a potential performance gain. What was interesting here is when I run the bench -count 10 it blows up[1].

```
cpu: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
                                                                │ without.txt  │              with_s.txt              │
                                                                │    sec/op    │    sec/op     vs base                │
TokenProgram/benchmark_token_program_compile_and_cache-12         2.595m ± ∞ ¹   2.575m ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_compile_and_cache_short-12   1.946m ± ∞ ¹   1.817m ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile-12                1.702m ± ∞ ¹   1.705m ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile_short-12          1.072m ± ∞ ¹   1.062m ± ∞ ¹       ~ (p=1.000 n=1) ²
geomean                                                           1.742m         1.706m        -2.07%
¹ need >= 6 samples for confidence interval at level 0.95
² need >= 4 samples to detect a difference at alpha level 0.05

                                                                │  without.txt  │               with_s.txt               │
                                                                │     B/op      │     B/op       vs base                 │
TokenProgram/benchmark_token_program_compile_and_cache-12         80.76Ki ± ∞ ¹   72.70Ki ± ∞ ¹        ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_compile_and_cache_short-12   47.44Ki ± ∞ ¹   39.55Ki ± ∞ ¹        ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile-12                264.4Ki ± ∞ ¹   240.8Ki ± ∞ ¹        ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile_short-12          231.3Ki ± ∞ ¹   207.3Ki ± ∞ ¹        ~ (p=1.000 n=1) ²
geomean                                                           123.7Ki         109.5Ki        -11.54%
¹ need >= 6 samples for confidence interval at level 0.95
² need >= 4 samples to detect a difference at alpha level 0.05

                                                                │ without.txt  │              with_s.txt              │
                                                                │  allocs/op   │  allocs/op    vs base                │
TokenProgram/benchmark_token_program_compile_and_cache-12         2.479k ± ∞ ¹   2.474k ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_compile_and_cache_short-12    395.0 ± ∞ ¹    414.0 ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile-12                2.471k ± ∞ ¹   2.472k ± ∞ ¹       ~ (p=1.000 n=1) ²
TokenProgram/benchmark_token_program_precompile_short-12           394.0 ± ∞ ¹    394.0 ± ∞ ¹       ~ (p=1.000 n=1) ³
geomean                                                            988.1          999.4        +1.14%
```

[1] 
> thread '<unnamed>' panicked at 'failed to spawn thread: Os { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }', /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/thread/mod.rs:686:29
stack backtrace:
   0:          0x168d351 - <unknown>
   1:          0x16f1adf - <unknown>
   2:          0x16814e7 - <unknown>
   3:          0x168d165 - <unknown>
   4:          0x168ff43 - <unknown>
   5:          0x168fcd4 - <unknown>
   6:          0x1690589 - <unknown>
   7:          0x1690487 - <unknown>
   8:          0x168d7b6 - <unknown>
   9:          0x16901d2 - <unknown>
  10:           0x7077b3 - <unknown>
  11:           0x707d43 - <unknown>
  12:          0x13b69f5 - <unknown>
  13:          0x139db4f - <unknown>
  14:          0x139928d - <unknown>
  15:          0x139912b - <unknown>
  16:          0x114657e - <unknown>
  17:           0xde920a - <unknown>
  18:           0xcbeb5f - <unknown>
  19:           0x7a1ce4 - <unknown>
fatal runtime error: failed to initiate panic, error 5
SIGABRT: abort
PC=0x7faad5c969fc m=6 sigcode=18446744073709551610
signal arrived during cgo execution